components/user/EditForm.vue

```diff
@@ -0,0 +1,294 @@
+    return (studyProgram: string) => {
+      const programme = studyProgrammes.value.find(
+        (prog: any) => prog.name === studyProgram
+      )
+
+      if (!programme || programme.type === 'integrated') return [1, 2, 3, 4, 5]
+
+      switch (programme.type) {
+        case 'bachelor':
+          return [1, 2, 3]
+        case 'master':
+          return [4, 5]
+        default:
+          return []
+      }
+    }
+  })
+
+  // get companies
+  const { data: companies } = await useFetch('/api/company')
+  const companyList = computed(() => {
+    return embedKeyIntoObjectValues(companies.value).map((company) => {
+      // eslint-disable-next-line @typescript-eslint/no-unused-vars
+      const { uid, name, ...remains } = company
+      return {
+        value: uid,
+        title: name,
+      }
+    })
+  })
+
+  // handle user privileges
+  const auth = useAuthStore()
+  const { hasAccess } = storeToRefs(auth)
+  const isAdmin = computed(() => hasAccess.value(['admin']))
+
+  const localePath = useLocalePath()
+
+  // Set state to user data plus companyUID
+  const state = reactive({ companyUID: null, ...props.user })
+
+  // Reset currentYear if programme doesn't have that year
+  watch(
+    () => state.studyProgram,
+    (program) => {
+      if (!yearOptions.value(program).includes(Number(state.currentYear)))
+        state.currentYear = ''
+    }
+  )
+
+  // check if user has changed
+  const hasChanged = computed(() => {
+    return !compareObjects(state, { companyUID: null, ...props.user })
+  })
+
+  // Alert state
+  const initialAlertState = {
+    show: false,
+    alertRoute: '',
+    type: undefined as AlertType,
+  }
+
+  const alertState = reactive({
+    ...initialAlertState,
+  })
+
+  // Show appropriate success alert after signing up for company
+  const displaySuccessAlert = (alertRoute: string) => {
+    alertState.alertRoute = alertRoute
+    alertState.type = 'success'
+    alertState.show = true
+  }
+
+  // show appropriate error alert after signing up for company
+  const displayErrorAlert = (alertRoute: string) => {
+    alertState.alertRoute = alertRoute
+    alertState.type = 'error'
+    alertState.show = true
+  }
+
+  // Show appropriate error alert for failed API calls
+  const displayErrorAlertFromMessage = (errorMessage: string) => {
+    const content = getAlertContent(errorMessage)
+    alertState.alertRoute = content.alertRoute
+    alertState.type = content.type
+    alertState.show = content.show
+  }
+
+  const form = ref()
+
+  // save changes to user
+  const saveChanges = async () => {
+    // throw error on invalid form
+    const { valid } = await form.value.validate()
+    try {
+      if (!valid) throw new Error('Form is not valid')
+    } catch (error) {
+      displayErrorAlert('alert.error.form.invalid')
+      return
+    }
+
+    if (state.userType !== 'company') {
+      state.companyUID = null
+    }
+
+    // update user
+    await useAsyncData('update-user', async () => {

```

Comment:
This is bloated as hell, we need to fix it at some point, but not now. See issue #285